tests/foreman/api/test_host.py

```diff
@@ -31,71 +34,61 @@
 from requests.exceptions import HTTPError
 
 from robottelo.api.utils import promote
-from robottelo.api.utils import publish_puppet_module
 from robottelo.config import settings
-from robottelo.constants import CUSTOM_PUPPET_REPO
 from robottelo.constants import ENVIRONMENT
 from robottelo.datafactory import invalid_interfaces_list
 from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
 from robottelo.datafactory import valid_hosts_list
 from robottelo.datafactory import valid_interfaces_list
 from robottelo.decorators import tier1
 from robottelo.decorators import tier2
 from robottelo.decorators import tier3
 from robottelo.decorators import upgrade
-from robottelo.decorators.func_locker import lock_function
-from robottelo.test import APITestCase
 
 
-class HostTestCase(APITestCase):
+class TestHost:
     """Tests for ``entities.Host().path()``."""
 
-    @classmethod
-    @lock_function
-    def setUpClass(cls):
-        """Setup common entities."""
-        super(HostTestCase, cls).setUpClass()
-        cls.org = entities.Organization().create()
-        cls.loc = entities.Location(organization=[cls.org]).create()
-        # Content View and repository related entities
-        cls.cv = publish_puppet_module(
-            [{'author': 'robottelo', 'name': 'generic_1'}],
-            CUSTOM_PUPPET_REPO,
-            organization_id=cls.org.id,
-        )
-        cls.env = (
+    @pytest.fixture(scope='module')
+    def module_env_search(self, module_org, module_location, module_cv_with_puppet_module):
+        env = (
             entities.Environment()
             .search(
                 query={
                     'search': 'content_view="{0}" and organization_id={1}'.format(
-                        cls.cv.name, cls.org.id
+                        module_cv_with_puppet_module.name, module_org.id
                     )
                 }
             )[0]
             .read()
         )
-        cls.env.location.append(cls.loc)
-        cls.env.update(['location'])
-        cls.lce = (
+        env.location.append(module_location)
+        env.update(['location'])
+        return env
+
+    @pytest.fixture(scope='module')
+    def module_lce_search(self, module_org, module_env_search):
```

Comment:
`module_env_search` is not being used in this fixture. Is that intentional ?

Reply:
Also same can be moved to api_fixtures or if there are many number of fixtures for host, you can create separate host_fixtures module.

Reply:
DONE.